### PR TITLE
Feature/adding errors report

### PR DESCRIPTION
**Pull request by gyuvaraj10**
Work is in progress

### CODE REVIEW

File: app/scripts/app.js [hunk 1]
```diff
@@ -29,6 +29,8 @@ angular
     localStorageServiceProvider.setPrefix('testReporter');
   }])
   .config(['$httpProvider', function ($httpProvider) {
+    $httpProvider.defaults.useXDomain = true;
```
Comment (sponte) [inline app/scripts/app.js]: We don't need this anymore as we're proxying requests through express/connect

File: app/scripts/controllers/view.js [hunk 2]
```diff
@@ -35,28 +48,45 @@ angular.module('testReporterApp')
           });
 
           var indexInSolr = function(testReport) {
+            console.log('Received test report', testReport.getHash(), testReport);
```
Comment (sponte) [inline app/scripts/controllers/view.js]: No console.log

File: app/scripts/controllers/view.js [hunk 3]
```diff
@@ -35,28 +48,45 @@ angular.module('testReporterApp')
...
             var solrReport = [];
-
+            var regexp = new RegExp('.*?\/testReport\/');
```
Comment (sponte) [inline app/scripts/controllers/view.js]: Unused

File: app/scripts/controllers/view.js [hunk 4]
```diff
@@ -35,28 +48,45 @@ angular.module('testReporterApp')
...
+                    time_to_live_s: '+1HOUR'
+                  };
+                  solrReport.push(document);
                 });
-              });
+              }
             });
 
-            $http.post('http://localhost:8983/solr/stats/update?commit=true', solrReport)
+            $http.get('/solr/stats/select?q=testReportId:' + testReport.getHash())
```
Comment (sponte) [inline app/scripts/controllers/view.js]: can this be moved to solr service please

File: app/scripts/modules/jenkins.js [hunk 5]
```diff
@@ -43,13 +43,13 @@
...
     //
-    this.id = (build.url + this.className + this.name).hashCode();
+    this.id = (build.url + this.className + this.name + execution.duration.toString()).hashCode();
 
     var urlComps = this.execution.className.split('.');
     var packageName = urlComps.splice(0, urlComps.length - 1).join('.') || '(root)';
     var className = urlComps[urlComps.length - 1];
 
-    this.url = this.build.url + 'testReport/' + packageName + '/' + className + "/" + testCaseUrlName(this.execution.name);
+    this.url = this.build.url + 'testReport/' + packageName + '/' + className.replace(/\//g, '_').replace(/^$/, '(empty)') + "/" + testUrlName(this.execution.name);
```
Comment (sponte) [inline app/scripts/modules/jenkins.js]: Refactor into a method please

File: app/views/view.html [hunk 6]
```diff
@@ -4,38 +4,98 @@ <h1>Listing for <span view-breadcrumbs="view.name"></span></h1>
...
-      job.displayName }}</a></td>
-    <td title="'Number of builds'" sortable="'builds.length'">{{ job.builds.length }}</td>
+      <test-report-table tests="search.testSearchResults"></test-report-table>
 
-    <td title="'Test pass rate'" sortable="'report.passRate'">{{ job.report.passRate | percentage }}</td>
-    <td title="'Unstable rate'" sortable="'report.unstableRate'">{{ job.report.unstableRate | percentage }}</td>
-    <td title="'Fail rate'" sortable="'report.failRate'">{{ job.report.failRate | percentage }}</td>
-    <td title="'Skip rate'" sortable="'report.skippedRate'">{{ job.report.skippedRate | percentage }}</td>
-    <td title="'Test count'" sortable="'report.cases.length'">{{ job.report.cases.length }}</td>
+      <!---->
```
Comment (sponte) [inline app/views/view.html]: Probably don't need these commented lines

File: gulpfile.js [hunk 7]
```diff
@@ -113,13 +126,14 @@ gulp.task('start:server', function() {
...
         return [
+          app,
           connect().use('/bower_components', connect.static('bower_components')),
           connect().use('/node_modules', connect.static('node_modules')),
           history()
         ];
     }
   });
-  browserSync.init({ proxy: 'localhost:9000' });
+  // browserSync.init({ proxy: 'localhost:9000' });
```
Comment (sponte) [inline gulpfile.js]: Put this back in please

**Review by sponte — CHANGES_REQUESTED** [review]
Please review these suggestions

File: app/scripts/app.js (hunk 1 above)
Comment (gyuvaraj10) [reply app/scripts/app.js]: Removed these lines

File: app/scripts/controllers/view.js (hunk 2 above)
Comment (gyuvaraj10) [reply app/scripts/controllers/view.js]: removed the console statements

File: app/scripts/controllers/view.js (hunk 3 above)
Comment (gyuvaraj10) [reply app/scripts/controllers/view.js]: removed the unsed regrex

**Pull request merged**